client/packages/system/src/Vaccination/Components/VaccinationModal.tsx

```diff
@@ -88,7 +103,17 @@ export const VaccinationModal = ({
       width={550}
       slideAnimation={false}
     >
-      {modalContent}
+      <>
+        {batchModalOpen && (
```

Comment:
I don't love the Modal over a Modal thing, but maybe can't be helped? Is it possible to show the batch list just as a Drop-down in the main modal, or is there too much info that needs to be squeezed in?

I think it's possible for a Drop-down item to be its own mini-components, so we could have the batch info formatted nicely in a little "card" within the drop-down selector?

Reply:
As discussed in Telegram - modal in modal was what was originally designed, but does digress from existing patterns... have gone back to product, I think would be easier to make it a dropdown if we can cull a column or two!

Will refactor in another PR if/when decision is made :)

Are you happy with these changes otherwise @CarlosNZ

Reply:
Sounds good, I'm happy to have a play with Drop-down options if/when we decide to do that.